src/main/java/org/broadinstitute/dsm/route/DownloadPDFRoute.java

```diff
@@ -197,235 +169,226 @@ public static Object returnPDFS(Map<String, Map<String, Object>> participantESDa
-                HttpServletResponse rawResponse = response.raw();
-                String fileName = "consent";
-                if (RELEASE_PDF.equals(pdfType)) {
-                    fileName = "release";
+        if (keySet.contains(JSON_END_DATE)) {
+            endDate = (String) jsonObject.get(JSON_END_DATE);
+            endDate = SystemUtil.changeDateFormat(SystemUtil.DATE_FORMAT, SystemUtil.US_DATE_FORMAT, endDate);
+        }
+        if (StringUtils.isBlank(medicalRecordId)) {
+            throw new RuntimeException("MedicalRecordID is missing. Can't create cover pdf");
+        }
+        //get information from db
+        MedicalRecord medicalRecord = MedicalRecord.getMedicalRecord(ddpInstance.getName(), ddpParticipantId, medicalRecordId);
+
+        PDFProcessor processor = new CoverPDFProcessor(ddpInstance.getName());
+        Map<String, Object> valueMap = new HashMap<>();
+        //values same no matter from where participant/institution data comes from
+        valueMap.put(CoverPDFProcessor.FIELD_CONFIRMED_INSTITUTION_NAME, medicalRecord.getName());
+        valueMap.put(CoverPDFProcessor.FIELD_CONFIRMED_INSTITUTION_NAME_2, medicalRecord.getName() + ",");
+        valueMap.put(CoverPDFProcessor.FIELD_CONFIRMED_FAX, medicalRecord.getFax());
+        String today = new SimpleDateFormat("MM/dd/yyyy").format(new Date());
+        valueMap.put(CoverPDFProcessor.FIELD_DATE, today);
+        valueMap.put(CoverPDFProcessor.FIELD_DATE_2, StringUtils.isNotBlank(endDate) ? endDate : today); //end date
+
+        //adding checkboxes configured under instance_settings
+        InstanceSettings instanceSettings = InstanceSettings.getInstanceSettings(ddpInstance.getName());
+        if (instanceSettings != null && instanceSettings.getMrCoverPdf() != null && !instanceSettings.getMrCoverPdf().isEmpty()) {
+            for (Value mrCoverSetting : instanceSettings.getMrCoverPdf()) {
+                if (keySet.contains(mrCoverSetting.getValue())) {
+                    valueMap.put(mrCoverSetting.getValue(), BooleanUtils.toBoolean((Boolean) jsonObject.get(mrCoverSetting.getValue())));
                 }
-                savePDFinBucket(ddpInstance.getName(), ddpParticipantId, new ByteArrayInputStream(bytes), fileName, userId);
-
-                rawResponse.getOutputStream().write(bytes);
-                rawResponse.setStatus(200);
-                rawResponse.getOutputStream().flush();
-                rawResponse.getOutputStream().close();
-            }
-            else {
-                throw new RuntimeException("Got null back for " + ddpParticipantId + " of ddpInstance " + ddpInstance.getName());
             }
         }
+        addDDPParticipantDataToValueMap(ddpInstance, ddpParticipantId, valueMap, true);
+
+        valueMap.put(CoverPDFProcessor.START_DATE_2, StringUtils.isNotBlank(startDate) ? startDate : valueMap.get(CoverPDFProcessor.FIELD_DATE_OF_DIAGNOSIS)); //start date
+        InputStream stream = null;
```

Comment:
You could use try-with-resource to work with auto closables (Input/Output streams) which will automatically close resources.
try( InputStream stream = processor.generateStream(valueMap) ) {
      //stuff do to here
} catch (Exception e) {
     //handling exception
}